### ISSUE TITLE
Recursive iterator detector false positive

### ISSUE THREAD

**Issue by rogercloud** [nim-lang/Nim#3354]
It's false positive when detecting iterator recursion. I only store one iterator copy in seq instead of calling it.

```Nim
type
    iterType = (iterator (): int {.closure.})

  var s = newSeq[iterType]()

  iterator iter(): int {.closure.}=
    s.add(iter)
    yield 1
```

````
Error: recursive dependency: 'iter'
```

**Comment by rogercloud** [comment]
I track the error and find the error in lambdalifting.nim line L866. The test case treciter.nim trigger semexprs.nim line L775. They're different errors. I totally understand the later one, but not the previoius. What's the reason to check the symbol here?

**Comment by kirbyfan64** [comment]
Fixed by 10a4bb6f4e52256f66dcffce472e808354ddad88?

**Comment by rogercloud** [comment]
Thanks, not sure if comments out the checker lines will cause other bugs... but it seems works in this case. I'll close the issue.
